crates/ra_hir/src/ty/tests.rs

```diff
@@ -586,6 +586,139 @@ fn test() -> i128 {
     );
 }
 
+#[test]
+fn infer_associated_const() {
+    check_inference(
+        "infer_associated_const",
+        r#"
+struct Struct;
+
+impl Struct {
+    const FOO: u32 = 1;
+}
+
+enum Enum;
+
+impl Enum {
+    const BAR: u32 = 2;
+}
+
+trait Trait {
+    const ID: u32;
+}
+
+struct TraitTest;
+
+impl Trait for TraitTest {
+    const ID: u32 = 5;
+}
+
+fn test() {
+    let x = Struct::FOO;
+    let y = Enum::BAR;
+    let z = TraitTest::ID;
```

Comment:
that this works is kind of accidental, because we currently just treat `impl Trait for Struct` as `impl Struct`... which is obviously a bug, but the results aren't that bad :smile:

Reply:
It's okay to leave these tests here even though `Const` will not be inferred yet ?

Reply:
Sure, we'll see the change when const support is added.